pkg/collectors/managedclusterinfo.go

```diff
@@ -0,0 +1,132 @@
+package collectors
+
+import (
+	"context"
+
+	ocinfrav1 "github.com/openshift/api/config/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/kube-state-metrics/pkg/metric"
+
+	mciv1beta1 "github.com/open-cluster-management/multicloud-operators-foundation/pkg/apis/internal.open-cluster-management.io/v1beta1"
+	"k8s.io/klog/v2"
+)
+
+var (
+	descClusterInfoName          = "ocm_managedcluster_info"
+	descClusterInfoHelp          = "Managed cluster information"
+	descClusterInfoDefaultLabels = []string{"hub_cluster_id", "cluster_id", "cluster", "vendor", "cloud", "version"}
+
+	cdGVR = schema.GroupVersionResource{
+		Group:    "hive.openshift.io",
+		Version:  "v1",
+		Resource: "clusterdeployments",
+	}
+
+	cvGVR = schema.GroupVersionResource{
+		Group:    "config.openshift.io",
+		Version:  "v1",
+		Resource: "clusterversions",
+	}
+
+	mciGVR = schema.GroupVersionResource{
+		Group:    "internal.open-cluster-management.io",
+		Version:  "v1beta1",
+		Resource: "managedclusterinfos",
+	}
+)
+
+func getHubClusterId(c dynamic.Interface) string {
+
+	cvObj, errCv := c.Resource(cvGVR).Get(context.TODO(), "version", metav1.GetOptions{})
+	if errCv != nil {
+		klog.Fatalf("Error getting cluster version %v \n", errCv)
+		panic(errCv.Error())
+	}
+	cv := &ocinfrav1.ClusterVersion{}
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(cvObj.UnstructuredContent(), &cv)
+	if err != nil {
+		klog.Fatalf("Error unmarshal cluster version object%v \n", err)
+		panic(errCv.Error())
+	}
+	return string(cv.Spec.ClusterID)
+}
+
+func getManagedClusterInfoMetricFamilies(client dynamic.Interface) []metric.FamilyGenerator {
+	hubClusterID := getHubClusterId(client)
+	return []metric.FamilyGenerator{
+		{
+			Name: descClusterInfoName,
+			Type: metric.MetricTypeGauge,
+			Help: descClusterInfoHelp,
+			GenerateFunc: wrapManagedClusterInfoFunc(func(mciObj *unstructured.Unstructured) metric.Family {
+				mci := &mciv1beta1.ManagedClusterInfo{}
+				err := runtime.DefaultUnstructuredConverter.FromUnstructured(mciObj.UnstructuredContent(), &mci)
+				if err != nil {
+					return metric.Family{Metrics: []*metric.Metric{}}
+				}
+				if mci.Status.ClusterID == "" ||
+					mci.Status.KubeVendor == "" ||
+					mci.Status.CloudVendor == "" ||
```

Comment:
we do not always return cloud vendor, what if it is baremetal?

Reply:
The CloudVendor is empty if baremetal or shows Other "CloudVendorOther"? 
What attribute shows that is baremetal?

Reply:
oh yeah, you are right. It return "Other" https://github.com/open-cluster-management/multicloud-operators-foundation/blob/60ef45157161b6d4ec2af817fb95d8e689a480e7/pkg/klusterlet/clusterinfo/clusterinfo_controller.go#L354